packages/@aws-cdk/aws-batch/lib/job-definition.ts

```diff
@@ -0,0 +1,345 @@
+     */
+    count: number;
+}
+
+/**
+ * Properties for a multi-node batch job
+ */
+export interface INodeRangeProps {
+    /**
+     * The container details for the node range.
+     */
+    container: IJobDefinitionContainer;
+    /**
+     * The minimum node index value to apply this container definition against. If omitted, the value zero is used.
+     * You may nest node ranges, for example 0:10 and 4:5, in which case the 4:5 range properties override the 0:10 properties.
+     */
+    fromNodeIndex?: number;
+    /**
+     * The maximum node index value to apply this container definition against. If omitted, the highest value is used relative
+     * to the number of nodes associated with the job. You may nest node ranges, for example 0:10 and 4:5,
+     * in which case the 4:5 range properties override the 0:10 properties.
+     */
+    toNodeIndex?: number;
+}
+
+/**
+ * Properties of a job definition
+ */
+export interface IJobDefinition extends IResource {
+    /**
+     * The ARN of this batch job definition
+     *
+     * @attribute
+     * @default Cloudformation-generated ARN
+     */
+    readonly jobDefinitionArn: string;
+
+    /**
+     * The name of the batch job definition
+     *
+     * @attribute
+     * @default Cloudformation-generated name
+     */
+    readonly jobDefinitionName: string;
+}
+
+/**
+ * Batch Job Definition
+ *
+ * Defines a batch job definition to execute a specific batch job.
+ */
+export class JobDefinition extends Resource implements IJobDefinition {
+    /**
+     * Fetches an existing batch job definition by its amazon resource name.
+     *
+     * @param scope
+     * @param id
+     * @param jobDefinitionArn
+     */
+    public static fromJobDefinitionArn(scope: Construct, id: string, jobDefinitionArn: string): IJobDefinition {
+        const stack = Stack.of(scope);
+        const jobDefName = stack.parseArn(jobDefinitionArn).resource;
+
+        class Import extends Resource implements IJobDefinition {
+            public readonly jobDefinitionArn = jobDefinitionArn;
+            public readonly jobDefinitionName = jobDefName;
+        }
+
+        return new Import(scope, id);
+    }
+
+    public readonly jobDefinitionArn: string;
+    public readonly jobDefinitionName: string;
+
+    constructor(scope: Construct, id: string, props: JobDefinitionProps = {}) {
+        super(scope, id, {
+            physicalName: props.jobDefinitionName,
+        });
+
+        const jobDef = new CfnJobDefinition(this, 'Resource', {
+            jobDefinitionName: props.jobDefinitionName,
+            containerProperties: this.buildJobContainer(props.containerProps),
+            type: 'container',
+            nodeProperties: props.nodeProps ? {
+                mainNode: props.nodeProps.mainNode,
+                nodeRangeProperties: this.buildNodeRangeProps(props.nodeProps),
+                numNodes: props.nodeProps.count,
+            } : undefined,
```

Comment:
Again, ternary operator indented not like in other places.

Reply:
Got it 👍